### PR TITLE
Unified Search

### PR DESCRIPTION
**Pull request by bitner**
Adds unified search using GeoNode Haystack and Registry located on the same elasticsearch instance.
Requires HAYSTACK_SEARCH to be setup and set to True, ES_UNIFIED_SEARCH = True and ES_URL and REGISTRY_URL to be set.

### CODE REVIEW

File: exchange/views.py [hunk 1]
```diff
@@ -127,6 +127,7 @@ def geoserver_reverse_proxy(request):
     return HttpResponse(req.content, content_type='application/xml')
 
 
+<<<<<<< 4d763fbbd0ea00c0dc23c983c1a6a7426eb9a57e
```
Comment (BerryDaniel) [inline exchange/views.py]: @bitner not sure how this even passed pep8, looks like you have a merge conflict artifacts L172 and L174 also.

File: requirements.txt [hunk 2]
```diff
@@ -22,3 +22,4 @@ django-flat-theme==1.1.3
 django-exchange-docs==1.1.2
 flake8==3.0.4
 git+git://github.com/boundlessgeo/django-osgeo-importer@b77ea961ea458b9c2ed8388ce7011a43e29f6fe9#egg=django-osgeo-importer
+elasticsearch-dsl==5.0.0
```
Comment (BerryDaniel) [inline requirements.txt]: @bitner is this a hard requirement? If so we need to get this to GEOINT ATO team asap <-- @davisc

Comment (bitner) [inline requirements.txt]: I could do without just using raw elasticsearch, that lib just makes easier

File: exchange/views.py (hunk 1 above)
Comment (bitner) [inline exchange/views.py]: argh, sorry ran flake8 before rebase

File: requirements.txt (hunk 2 above)
Comment (davisc) [inline requirements.txt]: @BerryDaniel lets update the spreadsheet to include this lib. @bitner can you resolve the conflict below?

**Pull request merged**